test/e2e/util_gatewayapi.go

```diff
@@ -0,0 +1,647 @@
+	if err := assertDNSRecord(t, dnsRecordName); err != nil {
+		return err
+	}
+
+	// Wait and check that the dns name resolves first. Takes a long time.
+	if err := wait.PollImmediate(20*time.Second, 5*time.Minute, func() (bool, error) {
+		_, err := net.LookupHost(hostname)
+		if err != nil {
+			t.Logf("waiting for HTTP route name %s to resolve...", hostname)
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		t.Fatalf("HTTP route name %s was unable to be resolved: %v", hostname, err)
+	}
+
+	// Wait for http route to respond, and when it does, check for the status code.
+	if err := wait.PollImmediate(5*time.Second, 5*time.Minute, func() (bool, error) {
+		statusCode, err := getHttpResponse(client, hostname)
+		if err != nil {
+			t.Logf("GET %s failed: %v, retrying...", hostname, err)
+			return false, nil
+		}
+		if statusCode != http.StatusOK {
+			t.Logf("GET %s failed: status %v, expected %v, retrying...", hostname, statusCode, http.StatusOK)
+			return false, nil // retry on 503 as pod/service may not be ready
+		}
+		t.Logf("request to %s was successful", hostname)
+		return true, nil
+
+	}); err != nil {
+		t.Fatalf("error contacting %s's endpoint: %v", hostname, err)
+	}
+
+	return nil
+}
+
+func getHttpResponse(client *http.Client, name string) (int, error) {
+	// Send the HTTP request.
+	response, err := client.Get("http://" + name)
+	if err != nil {
+		return 0, fmt.Errorf("GET %s failed: %v", name, err)
+	}
+
+	// Close response body.
+	defer response.Body.Close()
+
+	return response.StatusCode, nil
+}
+
+// assertCatalogSource checks if the CatalogSource of the given name exists,
+// and returns an error if not.
+func assertCatalogSource(t *testing.T, namespace, csName string) error {
+	t.Helper()
+	catalogSource := &operatorsv1alpha1.CatalogSource{}
+	nsName := types.NamespacedName{namespace, csName}
+
+	err := wait.PollImmediate(1*time.Second, 30*time.Second, func() (bool, error) {
+		if err := kclient.Get(context.TODO(), nsName, catalogSource); err != nil {
+			t.Logf("failed to get catalogSource %s: %v, retrying...", csName, err)
+			return false, nil
+		}
+		if catalogSource.Status.GRPCConnectionState != nil && catalogSource.Status.GRPCConnectionState.LastObservedState == "READY" {
+			t.Logf("found catalogSource %s with last observed state %s", catalogSource.Name, catalogSource.Status.GRPCConnectionState.LastObservedState)
+			return true, nil
+		}
+		t.Logf("found catalogSource %s but could not determine last observed state, retrying...", catalogSource.Name)
+		return false, nil
+	})
+	return err
+}
+
+// assertSMCP checks if the ServiceMeshControlPlane exists in a ready state,
+// and returns an error if not.
+func assertSMCP(t *testing.T) error {
+	t.Helper()
+	smcp := &maistrav2.ServiceMeshControlPlane{}
+	nsName := types.NamespacedName{openshiftIngressNamespace, openshiftSMCPName}
+
+	err := wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
```

Comment:
1 minute is pretty short for SMCP to come up. I usually see it take 2 minutes. I'd recommend 3-5 minutes to be safe.

Reply:
This is why I call this late 😄 , because it's always ready then. I can add the 2 minutes, not a problem.